### ISSUE TITLE
System crash when navigating back to first open page

### ISSUE THREAD

**Issue by dabega** [plentico/plenti#115]
When opening a page, navigating to a second one and going back to the first, the page is blank and any further interaction with the interface fails.

Steps to reproduce:
1- Go to https://plenti.co/
2- Click on any button (for instance "Give it to me")
3- Go back to previous page using browser navigation button

From what I've seen, it doesn't depend on the browser (tested on Chrome and Safari). It comes from the 'create_fragment' method because the 'ctx' variable is in this case mostly empty (ctx[1] is undefined and no verification in the code before trying to iterate over it). May be due to an issue with Svelte code itself, but didn't have time to go through it.

**Comment by jimafisk** [comment]
Wow good catch, I did not know about this. 

- https://plenti.co/ => any page => back :x: 
- https://plenti.co/ => any page => any page => back => back :x: 
- https://plenti.co/ => any page => https://plenti.co/ => any page => back => back => back :x: 
- https://plenti.co/themes => any page => back :heavy_check_mark: 
- https://plenti.co/docs => any page => back :heavy_check_mark: 
- https://plenti.co/download =>any page =>back :heavy_check_mark: 

Same thing happens with the default starter:
- http://localhost:3000/ => any page => back :x: 
- http://localhost:3000/ => any page => any page => back => back :x: 
- http://localhost:3000/ => any page => http://localhost:3000/ => any page => back => back => back :x: 
- http://localhost:3000/about => any page => back :heavy_check_mark: 
- http://localhost:3000/blog/stores => any page => back :heavy_check_mark: 

This appears to be an issue with the index page only, I assume it has something to do with my configuration of the client router and how a dynamic component is loaded. I'll look into it, thanks for flagging!

**Comment by dabega** [comment]
The crash itself which prevents further navigation on the site is easy enough to fix, but the cause (undefined values in ctx) needs to be addressed too since the page shows 'undefined' despite fixing the crash.

The fix is as follow:
Change {#each intro as paragraph} into {#each intro || [] as paragraph} in index.svelte

**Comment by jimafisk** [comment]
That's a clever way to prevent the error @dabega. When I do that, the `content` prop is still undefined when navigating back to the index:

<details>
<summary>Screenshot</summary>

![undefined](https://user-images.githubusercontent.com/5913244/113350411-698f8680-9307-11eb-80b0-b9198fac6904.png)

</details>

The problem seems to be that the `uri` variable is undefined when you navigate back to the homepage. If you `plenti eject` and choose the router, you can add a `console.log(uri);` right inside `function draw(m) {}` then load the homepage, go to the about page and use the browser's back button to return to the homepage and you'll see this:

<details>
<summary>Console screenshot</summary>

![uri_undefined](https://user-images.githubusercontent.com/5913244/113360422-c5620b80-9317-11eb-82cf-5ceb3df96942.png)


</details>

So it errors when we try to run: `content = getContent(uri);`

I'm not entirely sure why this is happening on the index page specifically... A quick fix that seems to work is simply setting a default value of `/` for the uri parameter in the `getContent` function:

```js
const getContent = (uri = "/", trailingSlash = "") => {                     
  return contentSource.find(content => content.path + trailingSlash == uri);
}
```

**Comment by jimafisk** [comment]
The root of the problem here is where we're trying to force the redirect from urls with a trailing slash to urls without a trailing slash:
```js
// Check if the url visited ends in a trailing slash (besides the homepage).
  if (uri.length > 1 && uri.slice(-1) == "/") {
    // Redirect to the same path without the trailing slash.
    router.route(content.path, false);
  }
```
Simply removing the `uri.length > 1 &&` fixes this issue and the redirects still work fine. I'll make this change for the next release.

**Comment by jimafisk** [comment]
Should be fixed in [v0.4.13](https://github.com/plentico/plenti/releases/tag/v0.4.13).

**Comment by jimafisk** [comment]
I broke the ability to use the browser's back button again in https://github.com/plentico/plenti/commit/3a8185fe98a0d7a617e54a907fb6986c0bcb6f4b (related to adding Baseurl support: https://github.com/plentico/plenti/issues/68).

**Comment by jimafisk** [comment]
This was fixed in [v0.4.22](https://github.com/plentico/plenti/releases/tag/v0.4.22).
